lesson14/script.js

```diff
@@ -0,0 +1,95 @@
+const showLoadingImg = () => {
+    const div        = document.createElement('div');
+    const loadingImg = document.createElement('img');
+    div.id           = "js-loading";
+    loadingImg.src   = "img/loading-circle.gif";
+    loadingImg.alt   = "ローディング画像";
+    document.body.insertAdjacentElement('afterbegin',div).appendChild(loadingImg);
+}
+
+const removeLoadingImg = () => {
+    document.getElementById('js-loading').remove();
+}
+
+const renderList = (result) => {
+    const fragment = document.createDocumentFragment();
+    for(const list of result){
+        const li  = document.createElement('li');
+        const a   = document.createElement('a');
+        const img = document.createElement('img');
+        li.className  = "list-item";
+        a.href        = `/${list.a}`;
+        a.textContent = list.text;
+        img.src       = list.img;
+        img.alt       = list.alt;
+        fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin',img);
+    } 
+    document.getElementById('js-list').appendChild(fragment);
+}
+
+const fetchdata = async () => {
+    try{ 
+    const endpoint = "https://api.json-generator.com/templates/szdgGQcOLXuk/data?access_token=hu4bc7qh9znx2m8f53mn4mz2hryvdntkavwbw8j0";
+    const response = await fetch(endpoint);
+    if(!response.ok){
+        console.error(`${response.status}:${response.statusText}`)
+    }
+    const json = await response.json();
+    const data = await json.data;
+    return data;
+    }catch(error){
+        console.error(error);
+    }
+}
+
+const init = async (number) => {
+    showLoadingImg();
+    try{
+        console.log(number);
+        const result = await fetchdata();
+        if(!result){
+            return
+        }
```

Comment:
I have learned a lot and I am happy to comment on this.
I was not using the if statement here and was getting a synctax error in the console if the value was false.
I learned how to work around it by looking at your code